packages/devtools_app/lib/src/screens/network/har_data_entry.dart

```diff
@@ -0,0 +1,283 @@
+      NetworkEventKeys.startedDateTime.name:
+          e.startTimestamp.toUtc().toIso8601String(),
+      NetworkEventKeys.time.name: e.duration?.inMilliseconds,
+      // Request
+      NetworkEventKeys.request.name: <String, Object?>{
+        NetworkEventKeys.method.name: e.method.toUpperCase(),
+        NetworkEventKeys.url.name: e.uri.toString(),
+        NetworkEventKeys.httpVersion.name: NetworkEventDefaults.httpVersion,
+        NetworkEventKeys.cookies.name: requestCookies,
+        NetworkEventKeys.headers.name: requestHeaders,
+        NetworkEventKeys.queryString.name: queryString,
+        NetworkEventKeys.postData.name: <String, Object?>{
+          NetworkEventKeys.mimeType.name: e.contentType,
+          NetworkEventKeys.text.name: e.requestBody,
+        },
+        NetworkEventKeys.headersSize.name:
+            _calculateHeadersSize(e.requestHeaders),
+        NetworkEventKeys.bodySize.name: _calculateBodySize(e.requestBody),
+      },
+      // Response
+      NetworkEventKeys.response.name: <String, Object?>{
+        NetworkEventKeys.status.name: e.status,
+        NetworkEventKeys.statusText.name: '',
+        NetworkEventKeys.httpVersion.name:
+            NetworkEventDefaults.responseHttpVersion,
+        NetworkEventKeys.cookies.name: responseCookies,
+        NetworkEventKeys.headers.name: responseHeaders,
+        NetworkEventKeys.content.name: <String, Object?>{
+          NetworkEventKeys.size.name: e.responseBody?.length,
+          NetworkEventKeys.mimeType.name: e.type,
+          NetworkEventKeys.text.name: e.responseBody,
+        },
+        NetworkEventKeys.redirectURL.name: '',
+        NetworkEventKeys.headersSize.name:
+            _calculateHeadersSize(e.responseHeaders),
+        NetworkEventKeys.bodySize.name: _calculateBodySize(e.responseBody),
+      },
+      // Cache
+      NetworkEventKeys.cache.name: <String, Object?>{},
+      NetworkEventKeys.timings.name: <String, Object?>{
+        NetworkEventKeys.blocked.name: NetworkEventDefaults.blocked,
+        NetworkEventKeys.dns.name: NetworkEventDefaults.dns,
+        NetworkEventKeys.connect.name: NetworkEventDefaults.connect,
+        NetworkEventKeys.send.name: NetworkEventDefaults.send,
+        NetworkEventKeys.wait.name: e.duration?.inMilliseconds ?? 0,
+        NetworkEventKeys.receive.name: NetworkEventDefaults.receive,
+        NetworkEventKeys.ssl.name: NetworkEventDefaults.ssl,
+      },
+      NetworkEventKeys.connection.name: e.hashCode.toString(),
+      NetworkEventKeys.comment.name: '',
+
+      // Custom fields
+      // har spec requires underscore to be added for custom fields, hence removing them
+      NetworkEventCustomFieldKeys.isolateId: '',
+      NetworkEventCustomFieldKeys.id: e.id,
+      NetworkEventCustomFieldKeys.startTime:
+          e.startTimestamp.microsecondsSinceEpoch,
+      NetworkEventCustomFieldKeys.events: [],
+    };
+  }
+
+  /// Converts the HAR data entry back to [DartIOHttpRequestData].
```

Comment:
nit: rephrase for accuracy "Returns the original [DartIOHttpRequestData] that this HAR entry was created from or parsed into."